fedimint-cli/src/ng.rs

```diff
@@ -196,29 +198,52 @@ pub async fn handle_ng_command(
         ClientNg::LnPay { bolt11 } => {
             client.select_active_gateway().await?;
 
-            let (operation_id, contract_id) = client.pay_bolt11_invoice(bolt11).await?;
-
-            let mut updates = client.subscribe_ln_pay(operation_id).await?.into_stream();
-
-            while let Some(update) = updates.next().await {
-                match update {
-                    LnPayState::Success { preimage } => {
-                        return Ok(serde_json::to_value(PayInvoiceResponse {
-                            operation_id,
-                            contract_id,
-                            preimage,
-                        })
-                        .unwrap());
+            let (pay_type, contract_id) = client.pay_bolt11_invoice(bolt11).await?;
+
+            match pay_type {
+                PayType::Internal(operation_id) => {
+                    let mut updates = client
+                        .subscribe_internal_pay(operation_id)
+                        .await?
+                        .into_stream();
+
+                    while let Some(update) = updates.next().await {
+                        if let InternalPayState::Preimage(preimage) = update {
```

Comment:
Is the success case the only state that we want the CLI to wait on? What happens if there's a refund?

Reply:
cli now reports refund, and also breaks the loop on error case